pkg/runner/registry.go

```diff
@@ -0,0 +1,76 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/testground/testground/pkg/api"
+	"github.com/testground/testground/pkg/aws"
+	"github.com/testground/testground/pkg/docker"
+	"github.com/testground/testground/pkg/rpc"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+)
+
+// pushToAWSRegistry takes a RunInput and pushes all images to the AWS ECR registry.
+// It then replaces the local artifact paths by the remote artifact paths.
+func pushToAWSRegistry(ctx context.Context, ow *rpc.OutputWriter, client *client.Client, in *api.RunInput) error {
+	ow.Infow("acquiring ECR authentication token")
+
+	// Get a Docker registry authentication token from AWS ECR.
+	auth, err := aws.ECR.GetAuthToken(in.EnvConfig.AWS)
```

Comment:
roughly equivilent to the aws commandline `aws ecr get-login` this gives us a temporary (12h) token to use with the docker client. We *could* split this up so one method of getting credentials is by aws token and another way is by commandline, etc. 

Although if we don't want to bother with other registries at this time, this is fine.

Reply:
I'd rather not fork a command-line process and deal with all that complexity, when we have an API that does the same as the CLI itself.

What registry is `kind` configured to use? Can we just add a registry type for `kind`? Is it using the local Docker daemon reigstry?